In [1]:
import pandas as pd

#CAC - Customer Acquisition Cost

##Calculate last month CRM expences

In [2]:
monthly_expences_sheet_id = '10OGbaywwMIqKgnPGy8VDvpBVtjyqln47iYa2lFhI9Mw'
monthly_expences = pd.read_excel(f'https://docs.google.com/spreadsheets/d/{monthly_expences_sheet_id}/export?format=xlsx', sheet_name='Sheet1')
monthly_expences.head()

#      month           category                  item  amount
0  1 2023-01-01       Server Costs           AWS Hosting    8000
1  2 2023-01-01       Server Costs  Google Cloud Storage    4000
2  3 2023-01-01  Software Licenses        Atlassian Jira    1200
3  4 2023-01-01  Software Licenses                 Slack     800
4  5 2023-01-01  Software Licenses            Salesforce    1500

In [3]:
last_month_expences = monthly_expences[monthly_expences['month'] == monthly_expences['month'].max()]
last_month_expences.head()

#      month           category                  item  amount
18  19 2023-03-01       Server Costs           AWS Hosting    8400
19  20 2023-03-01       Server Costs  Google Cloud Storage    4400
20  21 2023-03-01  Software Licenses        Atlassian Jira    1400
21  22 2023-03-01  Software Licenses                 Slack     900
22  23 2023-03-01  Software Licenses            Salesforce    1700

In [4]:
crm_expence = last_month_expences[last_month_expences['item']== 'Salesforce']['amount'].values[0]
crm_expence

np.int64(1700)

## Calculate last month Salaries

In [5]:
payroll_sheet_id = '1c_WihqTZCQvNgxzmd-OwhR9i5diwtfxXVLyMn8R-Lp4'
payroll_monthly = pd.read_excel(f'https://docs.google.com/spreadsheets/d/{payroll_sheet_id}/export?format=xlsx', sheet_name='Sheet1')
payroll_monthly.head()

month department  employee_name           position  paid
0 2023-01-01      Sales       John Doe      Sales Manager  1500
1 2023-01-01      Sales     Jane Smith    Sales Associate   600
2 2023-01-01      Sales      Jim Brown    Sales Associate   700
3 2023-01-01      Sales   Laura Miller    Sales Associate   800
4 2023-01-01  Marketing  Alice Johnson  Marketing Manager  1650

In [6]:
last_month_payroll = payroll_monthly[payroll_monthly['month'] == payroll_monthly['month'].max()]
last_month_payroll.head()

month department  employee_name           position  paid
34 2023-03-01      Sales       John Doe      Sales Manager  1500
35 2023-03-01      Sales     Jane Smith    Sales Associate   600
36 2023-03-01      Sales      Jim Brown    Sales Associate   700
37 2023-03-01      Sales   Laura Miller    Sales Associate   800
38 2023-03-01  Marketing  Alice Johnson  Marketing Manager  1650

We should calculate the payroll for marketing and sales since they are involve in the sell process. They include:
- Sales Associate
- Content Specialist

In [7]:
last_month_payroll_mkt_sale_expence = last_month_payroll[last_month_payroll['department'].isin(['Sales','Marketing'])]['paid'].sum()
last_month_payroll_mkt_sale_expence

np.int64(5950)

##Calculate last month Marketing costs

In [8]:
marketing_spending_sheet_id = '1AZOIThOV4P-0eYDge53ZwumVkfkHoYPWxst3k3Bv87c'
marketing_spendings = pd.read_excel(f'https://docs.google.com/spreadsheets/d/{marketing_spending_sheet_id}/export?format=xlsx', sheet_name='Sheet1')
marketing_spendings.head()

date       channel  spending
0 2023-01-01    Google Ads       784
1 2023-01-01  Facebook Ads       659
2 2023-01-01  LinkedIn Ads       729
3 2023-01-01   Twitter Ads       292
4 2023-01-02    Google Ads       935

In [9]:
last_month_marketing_expences = marketing_spendings[marketing_spendings['date'].dt.month == marketing_spendings['date'].max().month]['spending'].sum()
last_month_marketing_expences

np.int64(68830)

##Calculate last month Total Expences

In [10]:
TotalExpences = crm_expence + last_month_payroll_mkt_sale_expence + last_month_marketing_expences
TotalExpences

np.int64(76480)

##Calculate Number of Customers

In [11]:
receipts_history_sheet_id = '1qayqML1zCKdmtzutkcy9LWvE6xFRm6TGBEVkHHJKIuE'
receipts_history = pd.read_excel(f'https://docs.google.com/spreadsheets/d/{receipts_history_sheet_id}/export?format=xlsx', sheet_name='Sheet1')
receipts_history.head()

date  customer_id  receipt_amount  new_customer
0 2023-01-01         2653              67             1
1 2023-01-01         2731             271             1
2 2023-01-01         1277             231             0
3 2023-01-01         2094             107             0
4 2023-01-01         1314             416             0

In [12]:
last_month_new_customer = receipts_history[(receipts_history['date'].dt.month == receipts_history['date'].max().month)&(receipts_history['new_customer'] == 1)]
last_month_new_customer.head()

date  customer_id  receipt_amount  new_customer
621 2023-03-01         2406             426             1
622 2023-03-01         2761              41             1
625 2023-03-01         2844             252             1
627 2023-03-01         2679             323             1
633 2023-03-02         1475              40             1

In [13]:
number_customer = len(last_month_new_customer)
number_customer

63

##Final : Calculate the CAC

In [14]:
CAC = TotalExpences / number_customer
CAC

np.float64(1213.968253968254)

the CAC is about $1213.97

#Average Revenue Per User (ARPU)

In [15]:
#Calculate total revenue in the last month
total_revenue_last_month = receipts_history[receipts_history['date'].dt.month == receipts_history['date'].max().month]['receipt_amount'].sum()
total_revenue_last_month

np.int64(83033)

In [16]:
#Total customer in the last month
number_customer_last_month = receipts_history[receipts_history['date'].dt.month == receipts_history['date'].max().month]['customer_id'].nunique()
number_customer_last_month

292

In [17]:
#ARPU
ARPU = total_revenue_last_month / number_customer_last_month
ARPU

np.float64(284.3595890410959)

###Final: the ARPU is about 284

#COGS Cost of Goods Sold

Calculate everything spent on product development.(include: server cost and Atlassian Jira for technical support)

Part of Zoom and Slack cost should also be include. Since they were used by both dev team and back office
For simplicity, we assume 60% usage for COGS (dev team) and 40% usage for back office.



Production cost for server and software

In [18]:
production_items = ['AWS Hosting', 'Google Cloud Storage','Atlassian Jira']
last_month_production_expences = last_month_expences[last_month_expences['item'].isin(production_items)]['amount'].sum()

In [19]:
shared_software_items = ['Zoom','Slack']
last_month_shared_software_expences = last_month_expences[last_month_expences['item'].isin(shared_software_items)]['amount'].sum()

In [20]:
server_n_software_expences = last_month_production_expences + last_month_shared_software_expences * 0.6
server_n_software_expences

np.float64(15064.0)

In [21]:
last_month_production_salary = last_month_payroll[last_month_payroll['department']=='Engineering']['paid'].sum()
last_month_production_salary

np.int64(5200)

In [22]:
COGS = server_n_software_expences + last_month_production_salary
COGS

np.float64(20264.0)

#GROSS MARGIN

## Gross Margin calculation

In [23]:
GrossMargin = (total_revenue_last_month - COGS) / total_revenue_last_month
GrossMargin

np.float64(0.7559524526393121)

The result means Gross Margin is about 75.60%

##Gross Margin interpretation
- A 75% gross margin, or a 75% gross profit ratio, means that for every dollar of revenue, 75 cents is left over after accounting for the cost of producing the product or service.
- This remaining 75 cents can be used to cover operating expenses (like rent, salaries, marketing), and any remaining amount contributes to the company's net profit.
- For SaaS (Software as a Service) companies, a gross margin above 75% is often considered very good, indicating a healthy business model and efficient operations.
- A high gross margin also provides a business with more flexibility in terms of reinvesting in growth and managing unexpected costs.

In essence, a 75% gross margin suggests a company is effectively managing its production costs and has a strong foundation for profitability and growth.

#LTV Customer Lifetime Value

## Customer Lifespan calculation

In [24]:
lifespan_sheet_id = '1by8tPHwOnq3uKYK2E7sA9VBUYoPM4p1Rnrm_Ss9cyHI'
lifespan_data = pd.read_excel(f'https://docs.google.com/spreadsheets/d/{lifespan_sheet_id}/export?format=xlsx', sheet_name='Sheet1')
lifespan_data.head()

Unnamed: 0 start_date churn_date
0        1000 2021-11-15 2022-09-14
1        1001 2022-04-15 2023-02-16
2        1002 2022-10-30 2023-02-04
3        1003 2021-08-22 2023-02-07
4        1004 2021-08-23 2022-02-02

In [25]:
lifespan_data['lifespan_day'] = (lifespan_data['churn_date'] - lifespan_data['start_date']).dt.days
avg_lifespan_days = lifespan_data['lifespan_day'].mean()
avg_lifespan_month = (avg_lifespan_days / 30)
avg_lifespan_month

np.float64(9.841333333333333)

The customer average lifespan is around 9.84 months

##Final: LTV

In [26]:
LTV = ARPU * avg_lifespan_month * GrossMargin
LTV

np.float64(2115.5159315068495)

#LTV / CAC

In [27]:
LTV/CAC

np.float64(1.7426451841648996)

The "LTV/CAC" ratio is about 1.74

#Insights & Recommendation

#Summary of key values:
- Gross Margin (75.60%) : Excellent
- Profitability (Revenue > Expenses) : Positive
- Active customer acquisition (63 new customers)
- LTV (2,115.52) : good in general
- ARPU (284.36) : Healthy
- LTV/CAC: 1.74 : Ideally 3:1 or higher


##Key Insights:
**Profitability Under Pressure from Acquisition Costs:**

 While the company is profitable on paper this month and has an excellent gross margin, the low LTV/CAC ratio indicates that their customer acquisition strategy is not sustainable for long-term, scalable growth. They are spending too much to acquire customers compared to the value those customers bring.

**Marketing Efficiency is Low:**

  The significant marketing spend ($68,830) is not translating into a sufficiently high return on investment (as evidenced by the CAC and LTV/CAC). This suggests either:
*  Ineffective marketing channels/campaigns.
*  Targeting the wrong audience.
*  Poor conversion rates within the sales funnel.

**Retention Opportunity:**

A customer lifespan of under 10 months means churn is likely impacting overall LTV. Even a small improvement in retention could significantly boost LTV and, consequently, the LTV/CAC ratio.

**Production is Lean:**

The fact that COGS is entirely attributed to "Production salary" and the gross margin is high suggests that the core service delivery is efficient from a cost perspective.


##**Solutions & Recommendations:**

The primary focus should be on improving the LTV/CAC ratio. This can be achieved by:

**1. Reduce Customer Acquisition Cost (CAC):**

  **Optimize Marketing Channels:**
  - Analyze ROI per Channel: Identify which specific marketing channels and campaigns are yielding the best results (most new customers for the lowest cost) and reallocate budget away from underperforming ones.
  - A/B Test Ad Creatives and Messaging: Continuously test different ads, landing pages, and calls to action to improve conversion rates and lower cost per lead/acquisition.
  - Refine Targeting: Ensure marketing efforts are precisely targeted at the ideal customer profile to reduce wasted spend.
  - Explore Organic Growth: Invest more in SEO, content marketing, and community building to acquire customers at a lower or zero direct acquisition cost.

  **Improve Sales Efficiency:**
  - Sales Funnel Optimization: Identify bottlenecks in the sales process that lead to lost leads or extended sales cycles. Streamline processes.
  - Sales Training: Enhance sales team skills to improve conversion rates from qualified leads.
  - Leverage CRM Better: Ensure the CRM is being used effectively to manage leads, track interactions, and improve follow-up.

**2. Increase Lifetime Value (LTV):**

  - Improve Customer Retention (Lower Churn): This is often the most impactful area for LTV improvement.
  - Onboarding Optimization: Ensure new customers are successfully onboarded and quickly realize value from the product to reduce early churn.
  - Proactive Customer Success: Implement a robust customer success program to engage with users, provide support, address issues, and ensure they are getting the most out of the product.
  - Gather Feedback: Actively solicit customer feedback (surveys, interviews) to understand pain points and areas for product improvement that would increase satisfaction and retention.
  - Loyalty Programs/Incentives: Consider programs that reward long-term customers.
  - Increase Average Revenue Per User (ARPU):
Upselling/Cross-selling: Introduce higher-tier plans with more features, or complementary products/services, to existing customers.
  - Pricing Optimization: Review pricing strategy to ensure it aligns with the value provided and market expectations. Could there be room for a slight price increase for new customers, or value-based pricing?
  - Feature Adoption: Encourage customers to use more features of the product to deepen their reliance and perceived value.

## **Immediate Action Plan:**
- Deep Dive into Marketing Performance: Break down the $68,830 marketing expense by channel and campaign. Calculate the CAC for each. Identify the least efficient channels and either optimize them or reallocate budget.
- Churn Analysis: Understand why customers are churning around the 9-10 month mark. Are there common reasons? This will inform retention strategies.
- Customer Feedback Loop: Implement a systematic way to collect and act on customer feedback to improve product stickiness and satisfaction.

By focusing on these areas, the company can move towards a healthier LTV/CAC ratio, ensuring sustainable and profitable growth in the long run.
